lsp-metals.el

```diff
@@ -473,6 +473,25 @@ WORKSPACE is the workspace we received notification from."
                                    (add-hook 'lsp-on-idle-hook #'lsp-metals--did-focus nil t))
                   :completion-in-comments? t))
 
+(defmacro lsp-metals--create-bool-toggle (name config var)
+  "Create a toggle for lsp-metal config.
+NAME is a user-facing name used for the interactive command.  CONFIG is the LSP
+configuration name.  VAR is the variable holding the value of the configuration."
+  (let ((func-name (intern (format "lsp-metals-toggle-%s" name))))
+    `(defun ,func-name ()
+       ,(format "Toggle LSP metals %s config" name)
+      (interactive)
+      (setq ,var (not ,var))
```

Comment:
Lets put it in the workspace metadata, see https://github.com/emacs-lsp/lsp-metals/blob/51a89c1861eb505882c20393227f303ac33276e4/lsp-metals-treeview.el#L190
This way it will work correctly for multiple projects.